index.bs

```diff
@@ -8508,6 +8527,357 @@ worker=] algorithm:
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = {
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        }
+
+        storage.CookieFilter = {
+          ? ~ network.Cookie
+        }
+
+        storage.GetCookiesParameters = {
+          cookie: storage.CookieFilter,
+          ? partitionKey: storage.PartitionKey,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.GetCookiesResult = {
+        cookies: [*network.Cookie]
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for script.getCookies">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |filter| be the value of the <code>cookie</code> field of |command parameters|.
+1. Let |partition key spec| be the value of the <code>partitionKey</code> field of
+  |command parameters| if it is present or null if it isn't.
+1. Let |partition key| be the result of [=expand a storage partition key spec|expanding=] |partition key spec|.
+1. Let |store| be the result of [=trying=] to [=get the cookie store=] with
+  |partition key|.
+1. Let |cookies| be the result of [=get matching cookies=] with |store| and |filter|.
+1. Let |body| be a [=/map=] matching the <code>storage.GetCookiesResult</code> production,
+  with the <code>cookies</code> field set to |cookies|.
+1. Return [=success=] with data |body|.
+
+</div>
+
+#### The storage.setCookie Command ####  {#command-storage-setCookie}
+
+The <dfn export for=commands>storage.setCookie</dfn> command creates a new [=cookie=] in a cookie store, replacing any cookie in that store which matches according to [[COOKIES]].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.SetCookie = {
+          method: "storage.setCookie",
+          params: storage.SetCookieParameters,
+        }
+
+        storage.PartialCookie = {
+          name: text,
+          value: network.BytesValue,
+          domain: text,
+          ? ~ network.Cookie
+        }
+
+        storage.SetCookieParameters = {
+          cookie: storage.PartialCookie,
+          ? partitionKey: storage.PartitionKey,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.SetCookieResult = {
+        partitionKey: storage.partitionKey
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for script.setCookie">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |cookie spec| be the value of the <code>cookie</code> field of |command parameters|.
+1. If |cookie spec|["<code>size</code>"] exists:
```

Comment:
We should definitely change the CDDL instead of having this, but also in general we _don't_ raise errors for additional fields.

Reply:
Okay, I've made `Storage.CookieFilter` and `Storage.PartialCookie` extensible.